tests/src/Mod/Part/App/TopoShapeMakeShapeWithElementMap.cpp

```diff
@@ -105,3 +105,57 @@ TEST_F(TopoShapeMakeShapeWithElementMapTests, emptySourceShapes)
             &modifiedShape.makeShapeWithElementMap(source.getShape(), *Mapper(), emptySources));
     }
 }
+
+TEST_F(TopoShapeMakeShapeWithElementMapTests, nonMappableSources)
+{
+    // Arrange
+    auto [cube1, cube2] = TopoShapeExpansionHelpers::CreateTwoCubes();
+    std::vector<Part::TopoShape> sources {cube1, cube2};
+
+    // Act and assert
+    for (auto& source : sources) {
+        size_t canMap = 0;
+        for (const auto& mappableSource : sources) {
+            if (source.canMapElement(mappableSource)) {
+                ++canMap;
+            }
+        }
+
+        if (canMap == 0U) {
+            EXPECT_EQ(&source,
+                      &source.makeShapeWithElementMap(source.getShape(), *Mapper(), sources));
+        }
+    }
```

Comment:
The logic for this test is the same inside `makeShapeWithElementMap`.

Maybe there are smarter way to perform this test but as a first step copying the same logic should make the test useful.

Reply:
By definition the sources should be nonMappable, right?  So can you remove the inner loop and just go ahead and do the EXPECT_EQ?   This is really two tests right now - one of canMapElement, and the other of makeShapeWithElementMap.

Reply:
Yes, in theory the `mappableSources` as I defined them should return false when calling `canMapElement()`, therefor `canMap` should be ==0.

Calling `makeShapeWithElementMap()` in this condition should return `source` unmodified and this is what I want to test here.

If I do the same without the inner loop I guess this test would be more or less the same of `emptySourceShapes`